src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs

```diff
@@ -90,13 +90,38 @@ public MBuild.Project Project
                 {
                     return Context.ProjectCollection.LoadProject(FileInfo.FullName);
                 }
-                catch (InvalidProjectFileException ex)
+                catch (InvalidProjectFileException)
                 {
-                    throw new UpgradeException(LocalizedStrings.InvalidProjectError, ex);
+                    // TODO Delete obj file and retry - a change in platform version may cause load to fail for UWP/WinUI apps
+                    if (!DeleteObjDirectory())
+                    {
+                        throw;
+                    }
+
+                    try
+                    {
+                        return Context.ProjectCollection.LoadProject(FileInfo.FullName);
+                    }
+                    catch (InvalidProjectFileException e)
+                    {
+                        throw new UpgradeException(LocalizedStrings.InvalidProjectError, e);
+                    }
                 }
             }
         }
 
+        private bool DeleteObjDirectory()
+        {
+            var dirToDelete = FileInfo.Directory?.EnumerateDirectories().FirstOrDefault(directory => directory.Name == "obj");
```

Comment:
This feels fragile - for instance, this project redirects those directories to a something outside of the project directory.

What's the cause of the load failure? Is there a root cause that should be fixed in msbuild to handle it?

Reply:
When the `PlatformVersion` is upgraded (In the UpgradeTFM Step), and a dotnet restore is performed on the project, the `Context.ProjectCollection.LoadProject(FileInfo.FullName)` fails to load the project (I believe due to the inconsistencies in the files from older and newer version)

I could use the project property `BaseIntermediateOutputPath` to get the `obj` directory, but then that cannot be fetched until the project is loaded. So, it's kind of a chicken and egg situation and I have to resort to this solution.

If you have any suggestions, I would be happy to consider and implement it :)

Reply:
I identified specific files that cause the problem. 
Check if the new fix works instead 🙂